tests/integration/test_interaction.py

```diff
@@ -104,13 +199,19 @@ def parameterized_item_positive_feedback_on_bad_input(self, items_map, scroll_do
 
         for definition in self._get_correct_item_for_zone(items_map).values():
             if definition.input:
-                self.drag_item_to_zone(definition.item_id, definition.zone_id)
+                if use_keyboard:
```

Comment:
@itsjeyd nit: this if use_keyboard: ... else: ... is repeated multiple times without any changes - might make sense to extract into method. Also, you can use `if action_key is not None` to have three parameters instead of four